sympy/physics/control/lti.py

```diff
@@ -1226,224 +1179,909 @@ def get_asymptotic_stability_conditions(self, cancel_poles_zeros = False):
-        return self + other
+    TransferFunctionBase, TransferFunction, Feedback, Series, Parallel
 
-    def __sub__(self, other):
-        if hasattr(other, "is_StateSpace_object") and other.is_StateSpace_object:
-            return Parallel(self, -other)
-        elif isinstance(other, (TransferFunction, Series)):
-            if not self.var == other.var:
-                raise ValueError(filldedent("""
-                    All the transfer functions should use the same complex variable
-                    of the Laplace transform."""))
-            return Parallel(self, -other)
-        elif isinstance(other, Parallel):
-            if not self.var == other.var:
-                raise ValueError(filldedent("""
-                    All the transfer functions should use the same complex variable
-                    of the Laplace transform."""))
-            arg_list = [-i for i in list(other.args)]
-            return Parallel(self, *arg_list)
-        else:
-            raise ValueError("{} cannot be subtracted from a TransferFunction."
-                .format(type(other)))
+    """
+    def __new__(cls, num, den, var, sampling_time = 1):
+        if sampling_time == 0:
+            return TransferFunction(num, den, var)
 
-    def __rsub__(self, other):
-        return -self + other
+        sampling_time = sympify(sampling_time)
+        return super(DiscreteTransferFunction, cls).__new__(cls, num, den, var,
+                                                      sampling_time)
 
-    def __mul__(self, other):
-        if hasattr(other, "is_StateSpace_object") and other.is_StateSpace_object:
-            return Series(self, other)
-        elif isinstance(other, (TransferFunction, Parallel, Feedback)):
-            if not self.var == other.var:
-                raise ValueError(filldedent("""
-                    All the transfer functions should use the same complex variable
-                    of the Laplace transform."""))
-            return Series(self, other)
-        elif isinstance(other, Series):
-            if not self.var == other.var:
-                raise ValueError(filldedent("""
-                    All the transfer functions should use the same complex variable
-                    of the Laplace transform."""))
-            arg_list = list(other.args)
-            return Series(self, *arg_list)
-        else:
-            raise ValueError("TransferFunction cannot be multiplied with {}."
-                .format(type(other)))
+    def __init__(self, num, den, var, sampling_time = 1):
+        super().__init__()
+        self._sampling_time = sampling_time
 
-    __rmul__ = __mul__
+    @classmethod
+    def from_rational_expression(cls, expr, var=None, sampling_time = 1):
+        r"""
+        See :func:`TransferFunctionBase.from_rational_expression`.
+        """
+        return super().from_rational_expression(expr, var,
+                                         sampling_time = sampling_time)
```

Comment:
```suggestion
        return super().from_rational_expression(expr, var, sampling_time=sampling_time)
```

try not to have space around = on kwargs.